src/System.Diagnostics.StackTrace/src/System/Diagnostics/SymbolReader.CoreCLR.cs

```diff
@@ -0,0 +1,463 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+using System.Runtime.InteropServices;
+
+namespace System.Diagnostics
+{
+    public class SymbolReader
+    {
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        internal struct DebugInfo
+        {
+            public int lineNumber;
+            public int ilOffset;
+            public string fileName;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct MethodDebugInfo
+        {
+            public IntPtr points;
+            public int size;
+        }
+
+        /// <summary>
+        /// Read memory callback
+        /// </summary>
+        /// <returns>number of bytes read or 0 for error</returns>
+        internal unsafe delegate int ReadMemoryDelegate(IntPtr address, byte* buffer, int count);
+
+        /// <summary>
+        /// Internal holder for provider and reader cache.
+        /// </summary>
+        private struct HandleContents
+        {
+            readonly StackTraceSymbols _symbols;
+
+            public readonly MetadataReader PdbReader;
+
+            public HandleContents(StackTraceSymbols symbols, MetadataReader pdbReader)
+            {
+                _symbols = symbols;
+                PdbReader = pdbReader;
+            }
+
+            public void Dispose()
+            {
+                _symbols.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Stream implementation to read debugger target memory for in-memory PDBs
+        /// </summary>
+        private class TargetStream : Stream
```

Comment:
>TargetStream [](start = 22, length = 12)

This name + comment seems a little confusing. Is it fair to call this an InMemoryBufferStream? 
The comment mentions 'debugger target memory' which doesn't seem like an applicable concept here?